Places/View Controllers/LoginViewController.m

```diff
@@ -42,7 +43,9 @@ - (void)loginUser {
             NSLog(@"User logged in successfully");
             
             // display view controller that needs to shown after successful login
-            [self performSegueWithIdentifier:@"LoggedInSegue" sender:nil];
+            SceneDelegate *sceneDelegate = self.view.window.windowScene.delegate;
+            UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+            sceneDelegate.window.rootViewController = [storyboard instantiateViewControllerWithIdentifier:@"TabBarController"];
```

Comment:
It's usually preferable (and cleaner) to do this kind of thing through delegation. For instance, assuming `LoginViewController` is instantiated from `SceneDelegate`:

In `LoginViewController.h`:
```
@protocol LoginViewControllerDelegate <NSObject>
- (void)userDidLogin;
@end
@interface LoginViewController : UIViewController
...
@property (nonatomic, weak) id<LoginViewControllerDelegate> delegate;
@end
```
Then, in `LoginViewController.m`:
```
...
NSLog(@"User logged in successfully");
// call delegate after successful login
[self.delegate userDidLogin];
```
Finally, in `SceneDelegate.m`:
```
@interface SceneDelegate (Login) <LoginViewControllerDelegate>
@end
@implementation SceneDelegate (Login)
- (void)userDidLogin {
   // do whatever you need to do after the user logs in, i.e. replace the root view controller
}
@end
```